### PR TITLE
Don't use `master` as a branch name

### PR DESCRIPTION
**Pull request by erbridge**

### CODE REVIEW

File: rfc-036-use-main-as-the-default-branch-name.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+has its origin in master-slave relationships. It doesn't fit our ethics to
+perpetuate that language. It's an relatively small step, but by changing it we
+can take one of many steps needed to try to break the cycle of institutional
+racism.
+
+## Proposal
+
+Instead of calling the default branch of a Git repository `master` we SHOULD
+name it `main`. We MUST do this for all internal repositories, unless there's a
+more appropriate name for the default branch (such as `production`). We SHOULD
```
Comment (tahb) [inline rfc-036-use-main-as-the-default-branch-name.md:23]: On the point of "production" as an alternative name to "main". 

I don't intend to reignite a debate over branching models, whether it be one or many. The reality is that we do have many Rails projects that use multiple branches, usually in the form of a develop branch. I also believe that this is how all wordpress projects are set up by default.

I wonder if it would therefore be preferable to use "production" over "main" when used in conjunction with a "develop" branch. Is there a convention around the use of "main" that accounts for this situation that we could follow? Is this the done thing?

In short, do we prefer "Production/Develop" over "Main/Devleop"?

Comment (tahb) [reply rfc-036-use-main-as-the-default-branch-name.md]: I think it's okay to suggest some safe alternatives and point to the fact we need to make another decision on which one we want our default to be. 

My remaining concern is the next steps. As they are I think they risk us spending a lot of time changing our tooling and all _existing_ repositories (for example as a Friday task), only to make a subsequent decision that requires them all to be changed back from `main` to `release` (for example). I think the right thing to do is to stop using master everywhere but it feels more pragmatic to wait for the next RFC to be agreed on first. I think this RFC should be about not using master going forwards and the next RFC is about choosing the preferred name and taking corrective actions on past repos. 

> - Change all existing internal repositories to rename `master` to `main` or
  another more descriptive name
> - Change all template repositories to rename `master` to `main`

Comment (erbridge) [reply rfc-036-use-main-as-the-default-branch-name.md]: I've changed the next steps:

- Change all existing internal repositories with single "main" branches to
  rename `master` to `main`
- Change all template repositories to rename `master` to `main`
- Open another RFC to settle on naming conventions for repositories with
  multiple "main" branches (such as `master` and `develop`)

Does that work better?

**Comment by mallorydxw** [comment]
https://github.com/git/git/blob/83bdd4e509f59f9db364e9d5ada75e58423384d1/Documentation/config/init.txt#L5-L7

If you want to document how to do this, it looks like the `git` team are planning on calling the option `init.defaultBranch`. So the instructions to set this up would be just this command:

```
git config --global init.defaultBranch main
```

**Comment by erbridge** [comment]
Amazing. Do you know which version it will land in?

**Comment by jcoglan** [comment]
@erbridge It's still a work in progress and was first mentioned on the git dev mailing list about a week ago, so I'm not sure it's settled enough to know when it will ship.

**Comment by erbridge** [comment]
There's a note in the RFC about it. Thanks James.

**Comment by leeky** [comment]
FYI `init.defaultBranch` was included in the [Git v2.28 release](https://github.blog/2020-07-27-highlights-from-git-2-28/)
